source/includes/fundamentals/code-examples/Indexes.cs

```diff
@@ -0,0 +1,227 @@
+        // end-multi-key-query
+    }
+
+    private static void TextIndex(IMongoCollection<Movie> collection)
+    {
+        Console.WriteLine("text index");
+
+        try
+        {
+            // begin-text-index
+            collection.Indexes.CreateOne(Builders<Movie>.IndexKeys.Text("plot"));
+            // end-text-index
+        }
+        // Prints a message if a text index already exists with a different configuration 
+        catch (MongoCommandException e)
+        {
+            if (e.CodeName == "IndexOptionsConflict")
+            {
+                Console.WriteLine("There is an existing text index with different options");
+            }
+        }
+
+        // begin-text-query
+        // Define query parameters
+        var filter = Builders<Movie>.Filter.Text("java coffee shop");
+        var projection = Builders<Movie>.Projection.Include("plot").Exclude("_id");
+
+        // Execute query
+        var results = collection.Find(filter).Project(projection);
+        // end-text-query
+    }
+
+    private static void GeoSpatialIndex(IMongoCollection<Theater> collection)
+    {
+        Console.WriteLine("geospatial index");
+
+        try
+        {
+            // begin-geospatial-index
+            collection.Indexes.CreateOne(Builders<Theater>.IndexKeys.Geo2DSphere("location.geo"));
+            // end-geospatial-index
+        }
+        // Prints a message if a geospatial index already exists with a different configuration 
+        catch (MongoCommandException e)
+        {
+            if (e.CodeName == "IndexOptionsConflict")
+            {
+                Console.WriteLine("There is an existing geospatial index with different options");
+            }
+        }
+
+        // begin-geospatial-query
+        // Stores the coordinates of the NY MongoDB headquarters
+        var refPoint = GeoJson.Point(GeoJson.Position(-73.98456, 40.7612));
+
+        // Creates a filter to match documents that represent locations up to 1000 meters from the specified point directly from the geospatial index
+        var filter = Builders<Theater>.Filter.Near("location.geo", refPoint, 1000.0, 0.0);
+
+        // Execute the query
+        var results = collection.Find(filter);
+        // end-geospatial-query
+    }
+
+    private static void UniqueIndex(IMongoCollection<Theater> collection)
+    {
+        Console.WriteLine("unique index");
+
+        // begin-unique-index
+        var options = new CreateIndexOptions { Unique = true };
+        collection.Indexes.CreateOne(
+            Builders<Theater>.IndexKeys.Descending("theaterId"),
+            options);
+        // end-unique-index
+    }
+
+    private static void WildcardIndex(IMongoCollection<Theater> collection)
+    {
+        Console.WriteLine("wildcard index");
+
+        // begin-wildcard-index
+        collection.Indexes.CreateOne(Builders<Theater>.IndexKeys.Ascending("location.$**"));
```

Comment:
@JamesKovacs Wanted to specifically draw your attention here – is this the preferred way to implement an ascending wildcard index using the Builders model?

Reply:
We have a `Wildcard` method on `IndexKeys`.

```
coll.Indexes.CreateOne(
    new CreateIndexModel<Theater>(Builders<Review>.IndexKeys.Wildcard(x => x.Location))
);
```

Reply:
In this case, would the `x.Location` wildcard be defined in the `Theater` Poco as something like this?

```
        [BsonElement("location.$**")]
        public string LocationGeo { get; set; }
```

Reply:
`[BsonElement]` specifies the field name in the database and is unrelated to wildcard indexes. You would still specify this field as `location.geo`.